File: seriousGameBackend/src/main/java/no/ntnu/bachelor/models/Subject.java
package no.ntnu.bachelor.models;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

@Entity(name = "subjects")
public class Subject
{
    @Id
    @GeneratedValue
    private Long id;

    @OneToMany(mappedBy = "subject", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<Level> levels = new HashSet<>();

    private String subjectName;
    private String backgroundUrl;
    private String subjectDescription;

    public Subject() {}

    public Subject(String subjectName,  String backgroundUrl, String subjectDescription)
    {
        this.subjectName = subjectName;
        this.backgroundUrl = backgroundUrl;
        this.subjectDescription = subjectDescription;
    }

    @Override
    public String toString()
    {
        return String.format("Level[id=%d, subjectName='%s', backgroundUrl='%s', subjectDescription='%s']",
                id, subjectName, backgroundUrl, subjectDescription);
    }

    public Long getId() { return id; }

    public Set<Level> getLevels() {return levels;}

    public void setLevels(Set<Level> levels) { this.levels = levels; }

    public String getSubjectName() { return subjectName; }

    public void setSubjectName(String subjectName) { this.subjectName = subjectName; }

    public String getBackgroundUrl() { return backgroundUrl; }

    public void setBackgroundUrl(String backgroundUrl) { this.backgroundUrl = backgroundUrl; }

    public String getSubjectDescription() { return subjectDescription; }

    public void setSubjectDescription(String subjectDescription) { this.subjectDescription = subjectDescription; }
}

File: Bachelor/Assets/Scripts/Subject.cs
using System;
using System.Collections.Generic;
using UnityEngine;

[Serializable]
public class Subject
{
    public long ID
    {
        get => id;
        set => id = value;
    }

    public List<Level> Levels
    {
        get => levels;
        set => levels = value;
    }

    public string SubjectName
    {
        get => subjectName;
        set => subjectName = value;
    }

    public string BackgroundUrl
    {
        get => backgroundUrl;
        set => backgroundUrl = value;
    }

    public string SubjectDescription
    {
        get => subjectDescription;
        set => subjectDescription = value;
    }

    [SerializeField] private long id;
    [SerializeField] private List<Level> levels = new ();
    [SerializeField] private string subjectName;
    [SerializeField] private string backgroundUrl;
    [SerializeField] private string subjectDescription;
    
    public Subject() {}

    public Subject(string subjectName, string backgroundUrl, string subjectDescription)
    {
        this.subjectName = subjectName;
        this.backgroundUrl = backgroundUrl;
        this.subjectDescription = subjectDescription;
    }
}